packages/core/src/shared/fs/fs.ts

```diff
@@ -211,6 +222,37 @@ export class FileSystem {
             return nodefs.rename(oldUri.fsPath, newUri.fsPath).catch(errHandler)
         }
 
+        /**
+         * We were seeing 'FileNotFound' errors during renames, even though we did a `writeFile()` right before the rename.
+         * The error looks to be from here: https://github.com/microsoft/vscode/blob/09d5f4efc5089ce2fc5c8f6aeb51d728d7f4e758/src/vs/platform/files/node/diskFileSystemProvider.ts#L747
+         * So a guess is that the exists()(stat() under the hood) call needs to be retried since there may be a race condition.
+         */
+        let attempts = 0
+        const isExists = await waitUntil(async () => {
+            const result = await fs.exists(oldUri)
+            attempts += 1
+            return result
+        }, renameTimeoutOpts)
+        // TODO: Move the `aws_fileSystem` or some variation of this metric in to common telemetry
+        if (!isExists) {
+            // source path never existed after multiple attempts.
+            // Either the file never existed, or had we waited longer it would have. We won't know.
+            telemetry.aws_fileSystem.emit({
+                result: 'Failed',
+                action: 'rename',
+                reason: 'SourceNotExists',
+                reasonDesc: `After ${renameTimeoutOpts.timeout}ms the source path did not exist: ${oldUri.fsPath}`,
```

Comment:
does `oldUri.fsPath` need to be scrubbed https://github.com/aws/aws-toolkit-vscode/blob/5dfc8c59a1ea268f0bedb02f0fae5cf8356e01dd/packages/core/src/shared/errors.ts#L334

Reply:
Does it make sense for telemetry to process any `reasonDesc` through scrub names automatically?
We could search for `reasonDesc` [somewhere here](https://github.com/nkomonen-amazon/aws-toolkit-vscode/blob/8d88a04d1ebf4e14a7e87b94200d299278034a10/packages/core/src/shared/telemetry/spans.ts#L129) and process it through `scrubNames()` always.

But for this current code I'll just use scrubNames inline and we can revisit in a future PR